src/Controller/CRUDController.php

```diff
@@ -1359,15 +1362,37 @@ protected function handleXmlHttpRequestErrorResponse(Request $request, FormInter
             return $this->renderJson([], Response::HTTP_NOT_ACCEPTABLE);
         }
 
-        $errors = [];
-        foreach ($form->getErrors(true) as $error) {
-            $errors[] = $error->getMessage();
+        return $this->json(
+            $this->buildConstraintViolationList($form),
+            Response::HTTP_BAD_REQUEST
+        );
+    }
+
+    private function buildConstraintViolationList(FormInterface $form): ConstraintViolationList
+    {
+        $errors = new ConstraintViolationList();
+
+        foreach ($form->getErrors(true) as $formError) {
+            $formName = $this->buildName($formError->getOrigin());
+            $closure = \Closure::bind(function (ConstraintViolationInterface $violation) use ($formName): ConstraintViolationInterface {
+                $violation->propertyPath = $formName;
+
+                return $violation;
+            }, null, ConstraintViolation::class);
+
+            $errors->add($closure($formError->getCause()));
         }
 
-        return $this->renderJson([
-            'result' => 'error',
-            'errors' => $errors,
-        ], Response::HTTP_BAD_REQUEST);
+        return $errors;
+    }
+
+    private function buildName(FormInterface $form): string
```

Comment:
Again, the same with buildId, it can't be reused by symfony. It is built on `createView` thanks to: https://github.com/symfony/symfony/blob/6.2/src/Symfony/Component/Form/Extension/Core/Type/BaseType.php#L49